src/pages/sources/AddSourceModal.tsx

```diff
@@ -0,0 +1,223 @@
+import { ActionGroup, Button, Checkbox, DropdownItem, Form, FormContextProvider, FormGroup, HelperText, Modal, ModalVariant, NumberInput, TextArea, TextContent, TextInput } from '@patternfly/react-core';
+import * as React from 'react';
+import axios from 'axios';
+import { TypeaheadCheckboxes } from 'src/components/TypeaheadCheckboxes';
+import { SourceType } from 'src/types';
+import { SimpleDropdown } from 'src/components/SimpleDropdown';
+
+export interface AddSourceModalProps {
+    type: string;
+    onClose: () => void;
+    onSubmit: (payload) => void;
+}
+
+const AddSourceModal: React.FC<AddSourceModalProps> = ({
+    type,
+    onClose,
+    onSubmit
+}) => {
+    const [credOptions, setCredOptions] = React.useState<{ value: string, label: string }[]>([]);
+    const [credentials, setCredentials] = React.useState<string[]>([]);
+    const [useParamiko, setUseParamiko] = React.useState<boolean>(false);
+    const [sslVerify, setSslVerify] = React.useState<boolean>(true);
+    const [sslProtocol, setSslProtocol] = React.useState<string>('SSLv23');
+
+    const typeValue = type.split(' ').shift()?.toLowerCase();
+    const isNetwork = typeValue === "network";
+    // //example payload:
+    // const payload = {
+    //     "source_type": typeValue,
+    //     "credentials": credentials,
+    //     "hosts": [
+    //         "124.124.142"
+    //     ],
+    //     "name": "network source name",
+    //     "port": "23",
+    //     "options": {
+    //         "use_paramiko": true
+    //     }
+    // }
+    // //nonnetwork, url sources/?scan=true
+    // const payload2 = {
+    //     "source_type": typeValue,
+    //     "credentials": [
+    //         2
+    //     ],
+    //     "hosts": [
+    //         "123.123.123"
+    //     ],
+    //     "name": "vcenter source name",
+    //     "port": 443,
+    //     "options": {
+    //         "ssl_cert_verify": true,
+    //         "ssl_protocol": "SSLv23",
+    //         "disable_ssl": false
+    //     }
+    // }
+    // const payload3 = {
+    //     "source_type": "vcenter",
+    //     "credentials": [
+    //         1
+    //     ],
+    //     "hosts": [
+    //         "123.123.13"
+    //     ],
+    //     "name": "vcent 1271122",
+    //     "port": 443,
+    //     "options": {
+    //         "ssl_cert_verify": false,
+    //         "disable_ssl": true
+    //     }
+    // }
```

Comment:
I think it's okay having one example payload, but 3 feels like too much clutter.

I imagine you are using those to test things out, so my suggestion is to save these example payloads in a dedicated file.